### ISSUE TITLE
Auto-ETS parameter setting not working

### ISSUE THREAD

**Issue by kumar781** [sktime/sktime#5222]
**Describe the bug**
Want to set auto=False, damped_trend=True, seasonal= None and bound={damping_bound:(0,0.5)} for Auto-ETS-->

**To Reproduce**
model =AutoEts (auto=False, damped_trend = True, seasonal = None, bound={damping_bound:(0,0.5)}, sp=12, trend='add')

```python
model =AutoEts (auto=False, damped_trend = True, seasonal = None, bound={damping_bound:(0,0.5)}, sp=12, trend='add')
```

**Expected behavior**
I get the following error:
Arguments: (ValueError('Invalid key: smoothing_seasonal in bounds dictionary'),)

**Additional context**
Error: Arguments: (ValueError('Invalid key: smoothing_seasonal in bounds dictionary'),)

**Versions**
sktime 0.8.1, python 3.7
-->

</details>

**Comment by fkiraly** [comment]
Looks like you accidentally posted twice, the other issue is here?

https://github.com/sktime/sktime/issues/5221

**Comment by kumar781** [comment]
I have tried with sktime 0.13.2 and python 3.9
Still face the same issue

**Comment by kumar781** [comment]
Ideally I wanted 
model =AutoEts (auto=True, damped_trend = True, seasonal = None, bound={damping_bound:(0,0.5)}, sp=12, trend='add')
But it seems that if auto is set to True, you cannot manually provide a damping_bound range to the sktime auto-ets model.
Is there any workaround for this?

**Comment by kumar781** [comment]
def model_training(y_train):

            model_fcst = AutoETS(auto=True, additive_only=True, sp=12, trend='add', damped_trend=True,
                             allow_multiplicative_trend=False, bounds= {damping_bound:(0,0.5)}, error='add',
                              n_jobs=-1)
            model_fcst.fit(y_train)
            return model_fcst

**Comment by fkiraly** [comment]
is `sktime` 0.13.2 the most recent version you can use? We are at 0.22.0 currently.

If so, can you give a quick explanation why? Is there any compatibility restriction, or is it simply an internal pypi mirror?

**Comment by kumar781** [comment]
We are using 0.13.2 sktime because of indirect library version conflicts.
However, I have tried with sktime 0.22.0 too. Same issue there too.

**Comment by kumar781** [comment]
If this request is incorporated in the latest sktime release version, i can upgrade to it and find other ways to deal with library version conflicts or even kick it off in a separate docker container also.

**Comment by fkiraly** [comment]
Here's a self-contained piece of code that fails:

```python
from sktime.datasets import load_airline
from sktime.forecasting.ets import AutoETS

y_train = load_airline()
model_fcst = AutoETS(
    auto=False, additive_only=True, sp=12, trend='add', damped_trend=True,
    allow_multiplicative_trend=False, bounds= {"damping_bound":(0,0.5)}, error='add',
    n_jobs=-1
)
model_fcst.fit(y_train)
```

Reproduced on python 3.11, current main, under windows.

**Comment by kumar781** [comment]
# auto_ets_damping_trend_bound = {"damping_trend": (0, 0.8)}
from sktime.datasets import load_airline
from sktime.forecasting.ets import AutoETS
from sktime.forecasting.model_selection import temporal_train_test_split

# auto_ets_damping_trend_bound = {"damping_trend": (0, 0.8)}
auto_ets_damping_trend_bound = {"damping_trend": (0, 0.8), 'smoothing_seasonal': (0, 0.0), **{f"initial_seasonal.{i}": (0, 0.0) for i in range(12)}}
fh=12
data = load_airline()
y_train, y_test = temporal_train_test_split(data, test_size=12)

model_fcst = AutoETS(
    auto=False, additive_only=True, sp=12, trend='add', damped_trend=True,seasonal='add',
    allow_multiplicative_trend=False, bounds= auto_ets_damping_trend_bound, error='add',
    n_jobs=-1
)
model_fcst.fit(y_train)
model_fcst.update(y_test)
fcst=model_fcst.predict(fh=[1, 2, 3,4, 5, 6, 7, 8, 9, 10, 11, 12])
print(fcst)

**Comment by kumar781** [comment]
Please refer to the above working code. 
The feature we are requesting:

1. Ability to set auto=True and allow the model to automatically select best model from available hyper-parameters set such as seasonal='add' , damped_trend=True and manual bound provided.

Refer to the following code:

from sktime.datasets import load_airline
from sktime.forecasting.ets import AutoETS
from sktime.forecasting.model_selection import temporal_train_test_split

# auto_ets_damping_trend_bound = {"damping_trend": (0, 0.8)}
auto_ets_damping_trend_bound = {"damping_trend": (0, 0.8), 'smoothing_seasonal': (0, 0.0), **{f"initial_seasonal.{i}": (0, 0.0) for i in range(12)}}
fh=12
data = load_airline()
y_train, y_test = temporal_train_test_split(data, test_size=12)

model_fcst = AutoETS(
    auto=True, additive_only=True, sp=12, trend='add', damped_trend=True,seasonal='add',
    allow_multiplicative_trend=False, bounds= auto_ets_damping_trend_bound, error='add',
    n_jobs=-1
)

model_fcst.fit(y_train)

model_fcst.update(y_test)

fcst=model_fcst.predict(fh=[1, 2, 3,4, 5, 6, 7, 8, 9, 10, 11, 12])
print(fcst)

2. # auto_ets_damping_trend_bound = {"damping_trend": (0, 0.8)}
from sktime.datasets import load_airline
from sktime.forecasting.ets import AutoETS
from sktime.forecasting.model_selection import temporal_train_test_split

auto_ets_damping_trend_bound = {"damping_trend": (0, 0.8)}
# auto_ets_damping_trend_bound = {"damping_trend": (0, 0.8), 'smoothing_seasonal': (0, 0.0), **{f"initial_seasonal.{i}": (0, 0.0) for i in range(12)}}
fh=12
data = load_airline()
y_train, y_test = temporal_train_test_split(data, test_size=12)

model_fcst = AutoETS(
    auto=False, additive_only=True, sp=12, trend='add', damped_trend=True,seasonal=None,
    allow_multiplicative_trend=False, bounds= auto_ets_damping_trend_bound, error='add',
    n_jobs=-1
)

model_fcst.fit(y_train)

model_fcst.update(y_test)
fcst=model_fcst.predict(fh=[1, 2, 3,4, 5, 6, 7, 8, 9, 10, 11, 12])
fcst

Here it fails in the update block.

**Comment by fkiraly** [comment]
yes, thanks!
The bug is confirmed, it is waiting for someone to investigat and fix.

**Comment by kumar781** [comment]
Please let me know if you need any additional details about the requirement discussed above.

**Comment by kumar781** [comment]
Any updates on the bug? Will it be addressed in the upcoming sktime release?

**Comment by fkiraly** [comment]
Just looked into it, it does not seem to be a bug.
Kindly let us know if you think this is one, otherwise we will close the issue soon.

Why not a bug:

`damping_bound` is not a valid bound spec, per the `sktime` docs.

You probably mean `damping_trend`, this code works fine:

```python
from sktime.datasets import load_airline
from sktime.forecasting.ets import AutoETS

y_train = load_airline()
model_fcst = AutoETS(
    auto=False, additive_only=True, sp=12, trend='add', damped_trend=True,
    allow_multiplicative_trend=False, bounds= {"damping_trend":(0, 0.5)}, error='add',
    n_jobs=-1
)
model_fcst.fit(y_train)
```

**Comment by kumar781** [comment]
The main problem is I want to make auto=True and then choose from remaining from remaining settings like trend='add', 

    allow_multiplicative_trend=False, bounds= {"damping_trend":(0, 0.5)}, etc. In its current form if we make auto=True then it chooses from all defaults and disregards the specific hyper-parameter values i pass

**Comment by kumar781** [comment]
It is a bug because i am doing the same thing as you mentioned in your response. Only difference is i am creating a variable and calling it rather :
**auto_ets_damping_trend_bound = {"damping_trend": (0, 0.8)}**
fh=12
data = load_airline()
y_train, y_test = temporal_train_test_split(data, test_size=12)

model_fcst = AutoETS(
auto=False, additive_only=True, sp=12, trend='add', damped_trend=True,seasonal=None,
allow_multiplicative_trend=False, **bounds= auto_ets_damping_trend_bound,** error='add',
n_jobs=-1
)

model_fcst.fit(y_train)

**Comment by fkiraly** [comment]
Hm, @kumar781, could you kindly be explicit what your expectation is regarding the code you posted, i.e., what should happen, as compared to what actually happens?

**Comment by kumar781** [comment]
Currently what happens:
In auto-ets if I set auto=True, it doesn't acknowledge any other hyper-parameters I have set such as  additive_only=True, trend='add', damped_trend=True,seasonal=None, bounds, etc. It chooses the best model across all possibilities.

My expectation:
In auto-ets, if i set auto =True, then it should acknowledge all hyper-parameters that i have set and choose best model acknowledging my specifications like trend='add', seasonal=None, etc.

**Comment by kumar781** [comment]
If there's still confusion, we can schedule a call where we can discuss further as well. Thanks !

**Comment by fkiraly** [comment]
I understand the expectation now - let me check if I got it:

you set `auto=True` and a parameter `foo=42`, you expect the auto-ETS procedure to search only models with `foo=42`. However, currently, it seems to optimize `foo` too, for some parameters `foo` that can be set, such as `damped_trend`.

Additional question: how do you know that it also optimizes these parameters? What is the shortest way to verify this?

**Comment by kumar781** [comment]
Yes that's correct
I verified this by plotting graphs of forecasts with the auto=True and False. Additionally the official documentation of sktime also mentions the same

**Comment by kumar781** [comment]
I saw seasonality in forecast with auto=True and seasonality=False. Ideally in such a case I wanted the model to automatically search best model with seasonality =False. But the forecast graph was seasonal. This proved my intuition that setting auto=True does not acknowledge other hyper-parameter settings i choose.

**Comment by fkiraly** [comment]
Very observant, thanks.

So at least we should state clearly in the docstring what actually happens, and optimally see whether we can match the logic to the natural user expectation.

**Comment by kumar781** [comment]
I think optimally auto=True should acknowledge other hyper-parameters to choose best model. Its search space would be less and faster. I think it would be a great addition to sktime offerings.

**Comment by kumar781** [comment]
It would be great to know if sktime plans to add this new feature in its future release

**Comment by fkiraly** [comment]
> It would be great to know if sktime plans to add this new feature in its future release

Well, `sktime` has a "library" philosophy, i.e., we do not strictly control the estimators but allow authors to develop and maintain them as their own. For `AutoETS`, this would be a question to its current owner, @hyang1996.

There are two other options:

* someone else, perhaps even you, could contribute improvements to `AutoETS`!
* you could use the nixtla variant, `StatsForecastAutoETS` - you would have to check of course whether it respects set parameters as you would expect.

**Comment by kumar781** [comment]
It would be great to know your thoughts on this @HYang1996. Thanks in advance!

**Comment by HYang1996** [comment]
> It would be great to know your thoughts on this @HYang1996. Thanks in advance!

Hey @kumar781 , I haven't been an active contributor for a long time, so feel free to give it a go if you already know what to fix!

**Comment by fkiraly** [comment]
> I haven't been an active contributor for a long time

But this might be the golden opportunity for a glorious return!
(you're a former core dev so it's easy to "reactivate")

**Comment by kumar781** [comment]
I am using sktime make_reduction () for xgboost forecasting

Here is my working code:
Here is my code for xgboost forecasting using sktime:
from sktime.forecasting.model_selection import temporal_train_test_split
from sktime.forecasting.base import ForecastingHorizon
from sktime.forecasting.compose import make_reduction
from sktime.utils.plotting import plot_series
from sktime.performance_metrics.forecasting import mean_absolute_percentage_error

from sklearn.linear_model import LinearRegression

# Split data
y_train, y_test = temporal_train_test_split(y, test_size=26) # Predict from 1st July 2019

# Forecasting horizon, same as test data
fh = ForecastingHorizon(y_test.index, is_relative=False)

# Linear regression forecaster
regressor = LinearRegression()
forecaster = make_reduction(regressor, window_length=52, strategy="recursive")

# Fit and predict
forecaster.fit(y_train)
y_pred = forecaster.predict(fh)

# Plot predictions with training and test data
plot_series(y_train['2018-07-01':], y_test, y_pred, labels=["y_train", "y_test", "y_pred"], x_label='Date', y_label='Count pedestrians');

# Evaluate
print('MAPE: %.4f' % mean_absolute_percentage_error(y_test, y_pred, symmetric=False))

How do i specify "direct" in make_reduction() rather than 'recursive'. I get a bunch of errors if i specify strategy as direct
